static/app/views/performance/newTraceDetails/index.tsx

```diff
@@ -331,12 +335,32 @@ export function TraceViewWaterfall(props: TraceViewWaterfallProps) {
       return loadingTrace;
     }
 
-    if (props.trace) {
-      return TraceTree.FromTrace(props.trace, props.replayRecord);
+    if (trace) {
+      return TraceTree.FromTrace(trace, props.replayRecord);
     }
 
     throw new Error('Invalid trace state');
-  }, [props.traceSlug, props.trace, props.status, projects, props.replayRecord]);
+  }, [props.traceSlug, trace, isLoading, error, projects, props.replayRecord]);
+
+  useEffect(() => {
+    if (props.replayTraces) {
```

Comment:
You dont want to start fetching before the tree is assembled
```suggestion
    if (!props.replayTraces.length || tree.type !== 'trace') {
```

Reply:
I check for that inside the method

Reply:
Yes, but that is unnecessarily hidden. Just do it here and only call it if you need to